app/models/mission_control/jobs/workers/page.rb

```diff
@@ -0,0 +1,48 @@
+class MissionControl::Jobs::Workers::Page
+  DEFAULT_PAGE_SIZE = 10
+
+  attr_reader :workers_collection, :index, :page_size
+
+  def initialize(workers_collection, page: 1, page_size: DEFAULT_PAGE_SIZE)
+    @workers_collection = workers_collection
+    @page_size = page_size
+    @index = [ page, 1 ].max
+  end
+
+  def workers
+    workers_collection.drop(offset).first(page_size)
```

Comment:
I think it'd be better to assume here that `workers_collection` (or your generic collection) responds to `offset` and `limit`, in part because if this is an Active Record relation, calling `drop` and `first` over it will cause it to load the whole relation whereas using `offset` and `limit` will leverage these in the DB. 

You could create a light-weight version of `JobsRelation` for workers, where `offset` and `limit` are implemented, and have the server return that instead of calling `queue_adapter.workers` directly, very much like `ActiveJob::Querying.jobs` work. In that way you can use `limit` and `offset` here without worrying, and you'd have the basis for adding more filters to workers, in that `WorkersRelation`.